fe/fe-core/src/main/java/com/starrocks/sql/analyzer/InsertAnalyzer.java

```diff
@@ -335,4 +343,8 @@ private static ExternalOlapTable getOLAPExternalTableMeta(Database database, Ext
         }
         return copiedTable;
     }
+
+    public static boolean isUnSupportedPartitionColumnType(Type type) {
+        return type.isFloat() || type.isDecimalOfAnyVersion() || type.isDatetime();
```

Comment:
Double? array？map？struct？ I think it's better to check SupportedPartitionColumnType

Reply:
I will check these types in later patch.